kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/SniRoutingClusterNetworkAddressConfigProvider.java

```diff
@@ -113,41 +124,77 @@ public static class SniRoutingClusterNetworkAddressConfigProviderConfig {
 
         private final HostPort bootstrapAddress;
 
-        private final String brokerAddressPattern;
+        @JsonIgnore
+        private final String parsedBrokerAddressPattern;
         @JsonIgnore
         private final Pattern brokerAddressNodeIdCapturingRegex;
+        @JsonIgnore
+        private final Integer advertisedPort;
+        // present for serialize/deserialize fidelity
+        @SuppressWarnings("unused")
+        private final String brokerAddressPattern;
+        // present for serialize/deserialize fidelity
+        @SuppressWarnings("unused")
+        private final String advertisedBrokerAddressPattern;
 
         public SniRoutingClusterNetworkAddressConfigProviderConfig(@JsonProperty(required = true) HostPort bootstrapAddress,
-                                                                   @JsonProperty(required = true) String brokerAddressPattern) {
+                                                                   @Deprecated(forRemoval = true, since = "0.10.0") @JsonProperty String brokerAddressPattern,
```

Comment:
You could just use a JsonAlias on advertisedBrokerAddressPattern to retain support for the old name.  It means brokerAddressPattern would start to accept a port, but that's a backwardly compatibility change which won't break existing use-cases.

Reply:
`brokerAddressPattern` is already updated to accept a port in this PR. I had the same thought, it's a backwards-compatible change and easier to code if there's no behaviour difference.

This way is uglier, but it gives us a couple of things:

1. YAML serialize/deserialize fidelity, which I think we lean on for tests and now in the operator we're using these classes to generate broker config.
2. The deprecated annotation, helps generate warnings and find the thing to kill off later. A bit more tooling support than putting a comment by the alias.

Reply:
ah, and a chance to log a deprecation warning